homeassistant/components/system_log/__init__.py

```diff
@@ -186,14 +190,21 @@ def __init__(
         # This must be manually tested when changing the code.
         self.message = deque([_safe_get_message(record)], maxlen=5)
         self.exception = ""
-        self.root_cause: str | None = None
-        extracted_tb: traceback.StackSummary | None = None
+        self.root_cause: tuple[str, int, str] | None = None
+        extracted_tb: list[tuple[FrameType, int]] | None = None
         if record.exc_info:
-            self.exception = "".join(traceback.format_exception(*record.exc_info))
-            if extracted := traceback.extract_tb(record.exc_info[2]):
+            if formatter and record.exc_text is None:
+                record.exc_text = formatter.formatException(record.exc_info)
```

Comment:
This is unreachable under almost all circumstances since it will already be formatted in nearly all the cases.  Its still possible though if they add a log filter which suppresses the original logging